### PR TITLE
Blazor Hybrid routing and nav topic

### PR DESCRIPTION
**Pull request by guardrex**
Addresses #24956

[Internal Review Topic](https://review.docs.microsoft.com/en-us/aspnet/core/blazor/hybrid/routing?view=aspnetcore-6.0&branch=pr-en-us-25130)

Notes

* These will be *rough* at first. I'll circle around later for niceties like API doc cross-links (when available), enhanced descriptions, fully-working, cut-'n-paste code examples, etc.
* Although this PR doesn't include routing coverage, I think we'll probably be adding at least a brief remark that component routing (in a `BlazorWebView`) conforms to the coverage in the Blazor *Fundamentals* node topic on the subject with any caveats or *gotchas* 😈 that you want to call out. [*Fundamentals* topic on routing/nav](https://docs.microsoft.com/aspnet/core/blazor/fundamentals/routing) ... **_Does it all apply to Hybrid scenarios (`BlazorWebView`)?_**

### CODE REVIEW

**Comment by TanayParikh** [comment]
Thanks @guardrex. I'm going to hold off on reviewing till the original PR is merged, just in case there are minor changes like renaming the enum value.

File: aspnetcore/blazor/hybrid/routing.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+no-loc: ["Blazor Hybrid", Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: blazor/hybrid/routing
+---
+# ASP.NET Core Blazor Hybrid routing and navigation
+
+[!INCLUDE[](~/blazor/includes/blazor-hybrid-preview-notice.md)]
+
+Register to the `ExternalNavigationStarting` event and set the `ExternalLinkNavigationEventArgs.ExternalLinkNavigationPolicy` property to change link handling behavior. The `ExternalLinkNavigationPolicy` enumeration (`enum`) allows setting link handling behavior to `OpenInExternalBrowser`, `InsecureOpenInWebView`, and `CancelNavigation`. The `ExternalLinkNavigationEventArgs.Uri` property can be used to dynamically set link handling behavior.
+
+External links are opened in the device default browser by default. Opening external links within the Blazor WebView isn't recommended unless the content is fully trusted.
```
Comment (TanayParikh) [inline aspnetcore/blazor/hybrid/routing.md:18]:
```suggestion
> [!WARNING]
> External links are opened in the device default browser by default. Opening external links within the Blazor WebView can introduce security concerns and should not be enabled unless you can ensure all external links are fully trusted.
```

**Review by TanayParikh — APPROVED** [review]
LGTM other than one minor formatting change. Thanks!

File: aspnetcore/blazor/hybrid/routing.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## .NET MAUI
+
+Add the event handler to the constructor of the `Page` where the `BlazorWebView` is constructed:
+
+```csharp
```
Comment (TanayParikh) [inline aspnetcore/blazor/hybrid/routing.md:24]: Should we have something about adding a `using` statement for `Microsoft.AspNetCore.Components.WebView`?

File: aspnetcore/blazor/hybrid/routing.md [hunk 3]
```diff
@@ -15,20 +15,29 @@ uid: blazor/hybrid/routing
...
-    {
-        externalLinkNavigationEventArgs.ExternalLinkNavigationPolicy = 
-            ExternalLinkNavigationPolicy.InsecureOpenInWebView;
-    };
-```
+* Add the `Microsoft.AspNetCore.Components.WebView` namespace:
+
+  ```csharp
+  using Microsoft.AspNetCore.Components.WebView;
+  ```
```
Comment (TanayParikh) [inline aspnetcore/blazor/hybrid/routing.md:29]: This is required for all the platforms. Perhaps we should move this up above `## .NET MAUI` as a platform agnostic first step.

Comment (guardrex) [reply aspnetcore/blazor/hybrid/routing.md:29]: I'll patch it.

**Pull request merged**